tests/transaction_test_util.go

```diff
@@ -34,80 +34,157 @@ type TransactionTest struct {
 }
 
 type ttResult struct {
-	Byzantium      ttFork
-	Constantinople ttFork
-	Istanbul       ttFork
-	EIP150         ttFork
-	EIP158         ttFork
-	Frontier       ttFork
-	Homestead      ttFork
+	Prague         *ttFork
+	Cancun         *ttFork
+	Shanghai       *ttFork
+	Paris          *ttFork
+	London         *ttFork
+	Berlin         *ttFork
+	Byzantium      *ttFork
+	Constantinople *ttFork
+	Istanbul       *ttFork
+	EIP150         *ttFork
+	EIP158         *ttFork
+	Frontier       *ttFork
+	Homestead      *ttFork
 }
 
 type ttFork struct {
-	Sender common.UnprefixedAddress `json:"sender"`
-	Hash   common.UnprefixedHash    `json:"hash"`
+	Sender       *common.UnprefixedAddress `json:"sender"`
+	Hash         *common.UnprefixedHash    `json:"hash"`
+	Exception    *string                   `json:"exception"`
+	IntrinsicGas math.HexOrDecimal64       `json:"intrinsicGas"`
+}
+
+func (tt *TransactionTest) validate() error {
+	if tt.Txbytes == nil {
+		return fmt.Errorf("missing txbytes")
+	}
+	if err := tt.validateFork(tt.Result.Prague); err != nil {
```

Comment:
This should probably be a for loop

Reply:
If we iterate over a map `forks := map[string]ttFork` you can do `forks[Prague] = tt.Result.Prague`